crypto/dilithium/internal.h

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#ifndef AWSLC_HEADER_SIG_INTERNAL_H
+#define AWSLC_HEADER_SIG_INTERNAL_H
+
+#include <openssl/base.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+// NISTDSA_METHOD structure and helper functions.
```

Comment:
Happy to change the name -- originally wanted to use DSA but it was taken. I chose to add `NIST` because of the consistency between the FIPS 204, 205 standards, it seems as though any subsequent signature scheme standardizaed by NIST should follow these conventions. I didn't want to limit this to only signature schemes aimed towards PQ-secure use-cases.